cluster/gce/addons/apiserver-network-proxy/daemonset.yaml

```diff
@@ -0,0 +1,57 @@
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+    k8s-app: proxy-agent
+  namespace: kube-system
+  name: proxy-agent
+spec:
+  selector:
+    matchLabels:
+      k8s-app: proxy-agent
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        k8s-app: proxy-agent
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+    spec:
+      priorityClassName: system-cluster-critical
+      # Necessary to reboot node
+      hostPID: true
+      volumes:
```

Comment:
Is there a reason why we can't mount these via secret instead?

Reply:
My one concern is that I would like to minimize the dependencies of this feature on other systems. This needs to be working for things like log or admission webhooks to work. If we end up with a cyclic dependency, then you lose the ability to debug (log) and may lose the ability fix (fail close admission webhooks). It doesn't mean we can't use secrets but I do think we should be more cautious than we would normally be.

Reply:
I think secrets make a lot more sense, especially if we don't intend to run this on every node. We need to put the secret in a namespace that doesn't permit webhooks. I wonder if we should make a flag on namespace to override the webhook configurations and not call webhooks...